image-flash.c

```diff
@@ -120,6 +120,14 @@ static int flash_setup(struct image *image, cfg_t *cfg)
 		} else {
 			part->offset = partsize;
 		}
+		if (part->image) {
+			struct image *child = image_get(part->image);
+			if (child && child->size > part->size) {
```

Comment:
`!child` is an error. Please handle this the same way `hdimage_setup` does.

Reply:
Sure thing, fixed